amespahdbpythonsuite/mcfitted.py

```diff
@@ -0,0 +1,256 @@
+        self.__set(d, **keywords)
+
+    def set(self, d: Optional[dict] = None, **keywords) -> None:
+        """
+        Calls class: :class:`amespahdbpythonsuite.transitions.Transitions.set` to parse keywords.
+
+        """
+        Spectrum.set(self, d, **keywords)
+        self.__set(d, **keywords)
+
+    def __set(self, d: Optional[dict] = None, **keywords) -> None:
+        """
+        Populate data dictionary helper.
+
+        """
+        if isinstance(d, dict):
+            if d.get("type", "") == self.__class__.__name__:
+                if "observation" not in keywords:
+                    self.observation = d["observation"]
+                if "mcfits" not in keywords:
+                    self.mcfits = d["mcfits"]
+                if "mcweights" not in keywords:
+                    self.mcweights = d["mcweights"]
+                if "mcclasses" not in keywords:
+                    self.mcclasses = d["mcclasses"]
+                if "mcresults" not in keywords:
+                    self.mcresults = d["mcresults"]
+                if "mcpredicted" not in keywords:
+                    self.mcpredicted = d["mcpredicted"]
+                if "method" not in keywords:
+                    self.method = d["method"]
+
+        self.observation = keywords.get("observation", None)
+        self.mcfits = keywords.get("mcfits", list())
+        self.mcweights = keywords.get("mcweights", list())
+        self.mcclasses = keywords.get("mcclasses", list())
+        self.mcresults = keywords.get("mcresults", list())
+        self.mcpredicted = keywords.get("mcpredicted", False)
+        self.method = keywords.get("method", "")
+
+    def get(self) -> dict:
+        """
+        Calls class: :class:`amespahdbpythonsuite.transitions.Transitions.get`.
+        Assigns class variables from inherited dictionary.
+
+        """
+        d = Spectrum.get(self)
+        d["type"] = self.__class__.__name__
+        d["observation"] = self.observation
+        d["mcfits"] = self.mcfits
+        d["mcweights"] = self.mcweights
+        d["mcclasses"] = self.mcclasses
+        d["mcresults"] = self.mcresults
+        d["mcpredicted"] = self.mcpredicted
+        d["method"] = self.method
+
+        return d
+
+    def stats(self, results):
+        """
+        Obtain statistics (min, max, median, mean, std) for the MC fitted parameters,
+        and save to file.
+
+        """
+
+        # Save MC breakdown stats to file.
+        o = open('mcfitted_statistics.txt', 'w')
+        o.write('# param min max median mean std\n')
+
+        rkeys = list(results.keys())
+        for i, key in enumerate(rkeys):
+            o.write(f'{key} {np.min(results[key]):.5f} '
+                    f'{np.max(results[key]):.5f} '
+                    f'{np.median(results[key]):.5f} '
+                    f'{np.mean(results[key]):.5f} '
+                    f'{np.std(results[key]):.5f}\n')
+        o.close()
+
+    def averagespec(self, mcfits, mcclasses, **keywords):
```

Comment:
Conform Fitted, this should be the 'classes'-method, albeit now also return uncertainties.